core/domain/rights_manager.py

```diff
@@ -274,11 +340,19 @@ def _get_activity_rights_where_user_is_owner(activity_type, user_id):
         role.
     """
     if activity_type == constants.ACTIVITY_TYPE_EXPLORATION:
-        rights_model_class = exp_models.ExplorationRightsModel
+        rights_model_class: Union[
```

Comment:
Same comments as above regarding simplification and else clause.

Reply:
`rights_model_class` is removed and now we are defining models in if/else clause. 

Also, this method is only called in two places and there we are not providing `activity_type` other than these two.
https://github.com/oppia/oppia/blob/5f1b2e12f61f3daf750c44cc5dab58071ad582aa/core/domain/rights_manager.py#L292
https://github.com/oppia/oppia/blob/5f1b2e12f61f3daf750c44cc5dab58071ad582aa/core/domain/rights_manager.py#L307

So, i think defining expectation here is redundant !!

Reply:
See comment above regarding the exception and how to think about this sort of thing.

Reply:
assertion is added!